examples/vqe.qasm

```diff
@@ -59,7 +59,7 @@ def trial_circuit qubit[n]:q {
 /* Apply VQE ansatz circuit and measure a Pauli operator
  * given by spec. Return the number of 1 outcomes.
  */
-def counts_for_term(bit[2*n]:spec) qubit[n] -> uint[prec] {
+def counts_for_term(bit[2*n]:spec) qubit[n]:q -> uint[prec] {
```

Comment:
I think these examples have additional errors. I do not believe we support calling subroutines like gates, and the qubit `qubit: q` parameter should be included within the parentheses. The notation for specifying the size of the qubit array seems incorrect as well. I believe it should be something like:
```suggestion
def counts_for_term(bit[2*n]:spec, qubit[]:q) -> uint[prec] {
```
For now `n` is a constant that can be used within the subroutine for iteration bounds, however, the language should likely support a `length/sizeof` subroutine builtin for inquiring the size of an array.

Reply:
@taalexander do you mean that [the description of subroutine](https://qiskit.github.io/openqasm/language/subroutines.html#subroutines) will be changed? An example `xcheck` looks like a gate.
```
def xcheck qubit[4]:d, qubit:a -> bit {
  reset a;
  for i in [0: 3] cx d[i], a;
  return measure a;
}
```

Reply:
I believe we resolved that qubit args are not in the parentheses--see https://github.com/Qiskit/openqasm/pull/115/files.